### ISSUE TITLE
LCP-protected EPUBs fail to load because of LSD-related error

### ISSUE THREAD

**Issue by danielweck** [readium/readium-desktop#324]
Test with any of the 3 protected publications from:
https://www.edrlab.org/readium-lcp/testing-readium-lcp-compliant-devices/
(LCP passphrase "edrlab rocks")

Direct LCP license links:
* https://www.edrlab.org/public/readium-lcp/wasteland.lcpl
* https://www.edrlab.org/public/readium-lcp/accepub3.lcpl
* https://www.edrlab.org/public/readium-lcp/haruko.lcpl

Wasteland is a small publication, so I am attaching it here.

[wasteland_lcp.zip](https://github.com/readium/readium-desktop/files/3308035/wasteland_lcp.zip)

The following error is raised when attempting to open `wasteland.lcpl.epub` directly (i.e. not via the LCP license `wasteland.lcpl`): 

```
[2] ### LCP publication
[2] TypeError: Cannot read property 'lsd' of undefined
[2]     at LcpManager.<anonymous> (/Volumes/500GB/Code/readium-desktop/dist/main.js:5619:89)
[2]     at Generator.next (<anonymous>)
[2]     at /Volumes/500GB/Code/readium-desktop/dist/main.js:5490:71
[2]     at new Promise (<anonymous>)
[2]     at module.exports../src/main/services/lcp.ts.__awaiter (/Volumes/500GB/Code/readium-desktop/dist/main.js:5486:12)
[2]     at LcpManager.getLsdStatus (/Volumes/500GB/Code/readium-desktop/dist/main.js:5617:16)
[2]     at runCallEffect (/Volumes/500GB/Code/readium-desktop/node_modules/redux-saga/lib/internal/proc.js:524:19)
[2]     at runEffect (/Volumes/500GB/Code/readium-desktop/node_modules/redux-saga/lib/internal/proc.js:446:402)
[2]     at next (/Volumes/500GB/Code/readium-desktop/node_modules/redux-saga/lib/internal/proc.js:326:9)
[2]     at currCb (/Volumes/500GB/Code/readium-desktop/node_modules/redux-saga/lib/internal/proc.js:399:7)
[2]     at process._tickCallback (internal/process/next_tick.js:68:7)
[2] ### action type STREAMER_PUBLICATION_OPEN_ERROR
[2] ### action type READER_OPEN_ERROR
```

By rollback-ing the Git revision I managed to identify the culprit to:
https://github.com/readium/readium-desktop/commit/dedaae921ec3212ad361972bc91a0f8aaddca0dd

In other words, the commit immediately before in the history, works:
https://github.com/readium/readium-desktop/commit/937de9446e01ba9de2c0bb754d41e17669751110

(this makes sense, as there is a new check when attempting to load a publication)

**Comment by danielweck** [comment]
I suspect that `publicationDocument.lcp` is undefined when an LCP-protected publication is *not opened* from an LCPL license, so `publicationDocument.lcp.lsd.statusUrl` fails with an exception.

https://github.com/readium/readium-desktop/blob/42a6ab3efa20879db9f8f4b6503e15551b93df19/src/main/services/lcp.ts#L229-L236

**Comment by danielweck** [comment]
The second test is to open the LCPL license, let the app download the encrypted EPUB (and inject the `META-INF/license.lcpl`) ... then the book loads fine the first time. However subsequent times fail with `LICENSE_SIGNATURE_DATE_INVALID` / `111`, which indicates some error in how the LCP license is stored in the database. In fact, I can of course copy the `book.epub` from the private app storage, and attempt to load the copied EPUB => this results in the same `TypeError: Cannot read property 'lsd' of undefined`.

https://github.com/readium/readium-desktop/blob/42a6ab3efa20879db9f8f4b6503e15551b93df19/src/main/services/lcp.ts#L95-L97

**Comment by danielweck** [comment]
Regarding the `111` error, this actually seems legit, because the following chain of URLs does return a different LCPL from the original:

https://www.edrlab.org/public/readium-lcp/wasteland.lcpl

=>

https://lsd-prod.edrlab.org/licenses/9679f657-4a90-4e39-8f35-63784e269e6c/status

=>

https://front-prod.edrlab.org/api/v1/licenses/9679f657-4a90-4e39-8f35-63784e269e6c

(in fact, requesting the above URL multiple times never returns the same LCPL license ... not sure if this is a feature or a bug)

**Comment by danielweck** [comment]
But, the `111` should not happen in Thorium with Wasteland, because the LSD does not announce the availability of a more recent LCP license (so the original LCP license should always be in use). To be verified.
